src/main/scala/mesosphere/marathon/api/HTTPMetricsFilter.scala

```diff
@@ -0,0 +1,119 @@
+package mesosphere.marathon
+package api
+
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletRequestWrapper, HttpServletResponse, HttpServletResponseWrapper}
+import mesosphere.marathon.metrics.{Metrics, ServiceMetric}
+
+/**
+  * This filter replaces the default I/O streams with proxies that count
+  * the number of bytes that are sent to and from the client.
+  */
+class HTTPMetricsFilter extends Filter {
+  private[this] val inputBytesMetric = Metrics.counter(ServiceMetric, getClass, "inputBytes")
+  private[this] val outputBytesMetric = Metrics.counter(ServiceMetric, getClass, "outputBytes")
+
+  /**
+    * Wraps a `ServletOutputStream` and overrides the `write` method in
+    * order to count the number of bytes that went through the stream.
+    *
+    * @param upstream The stream to proxy the output to
+    */
+  private class OutputStreamCounter(val upstream: ServletOutputStream) extends ServletOutputStream {
```

Comment:
Well, I hate to break this to you but at least Apache Commons IO has this features: https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/CountingInputStream.html.

Reply:
But the object I have to wrap is not an `OuptuStream`, it's a `ServletOutputStream` :trollface:

Reply:
Hm, there must be an easier way. `ServletOutputStream` is actually not doing much.

Reply:
Ah man. I was so hopeful ^^.

Reply:
@jeschkies these other methods are necessary for async IO. See https://github.com/mesosphere/marathon/pull/6193/files#diff-52da5dcde3d226b130f1eff2c1c29c2b